### ISSUE TITLE
Unable to Knit or Publish on RS Connect

### ISSUE THREAD

**Issue by dareneiri** [sol-eng/connect-usage#13]
I am interested in at least trying to get this to run the example in this repo (rsc-usage.Rmd) with the scheduler in RS Connect. However, I have issues deploying the content to RS Connect from RS Server Pro. 

When I deploy to RS Connect, I see the following: 

> The CONNECT_SERVER and CONNECT_API_KEY environment variables are required in order for this report to pull usage data.
> 
> Please define these variables and then re-run the report. 

Of course, this is after adding the environment variables in RS Connect. 

If I try to Knit the document, it fails. 

> output file: rsc-usage.knit.md
> 
> /usr/lib/rstudio-server/bin/pandoc/pandoc +RTS -K512m -RTS rsc-usage.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output rsc-usage.html --email-obfuscation none --self-contained --standalone --section-divs --template /opt/R/3.6.2/lib/R/library/flexdashboard/rmarkdown/templates/flex_dashboard/resources/default.html --variable 'theme:cosmo' --include-in-header /tmp/RtmpGluOsU/rmarkdown-str9375782c95ff.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --lua-filter /opt/R/3.6.2/lib/R/library/rmarkdown/rmd/lua/pagebreak.lua --lua-filter /opt/R/3.6.2/lib/R/library/rmarkdown/rmd/lua/latex-div.lua --include-in-header /tmp/RtmpGluOsU/file937524480922html --include-before-body /tmp/RtmpGluOsU/file93753c40b650.html --highlight-style pygments --include-before-body /tmp/RtmpGluOsU/file9375432d15e8.html --include-after-body /tmp/RtmpGluOsU/file9375255c61be.html --css styles.css 
> pandoc: /tmp/RtmpGluOsU/email93758ba12e6_files/figure-html/add_plot-1.mb.png: openBinaryFile: does not exist (No such file or directory)
> Error: pandoc document conversion failed with error 1

If I just run the cells manually, they all work. I can even see the email report preview when I run the last cell. So the code runs, but there's some issue with deployment that I can't pin point. 

Any thoughts?

**Comment by slopp** [comment]
Sorry for the trouble!

@dareneiri are you seeing the error in knitting locally or on RStudio Connect? We'd expect the first error (about missing env variables) the first time you deploy, but by adding and saving those environment variables you should be able to proceed with rendering.

The rendering error is something I've seen before with older versions of markdown / blastula, could you try updating the packages? Alternatively, if you primarily want to see the report and are less concerned with the email, you could comment out the last code chunk that sets up the email and try rendering / deploying again.

**Comment by dareneiri** [comment]
Appreciate the quick response!!

I am seeing the knitting error locally. 

It does look like commenting out the last code chunk that sets up the email resolves the deployment issue in RS Connect. Once I added the variables in RS Connect, the report reloaded as expected. Previously, I'm assuming it tried to run that last code chunk but failed. 

I also updated from R 3.6.2 to 4.0.1 and updated the relevant packages. But I'm pretty certain it was that last code chunk that was causing me issues with knitting locally and deployment to RS Connect. 

At least this is progress. Thanks!
